signac/__main__.py

```diff
@@ -3,379 +3,192 @@
-        try:
-            job = _open_job_by_id(project, job_id)
-            job.move(dst_project)
-        except DestinationExistsError:
-            _print_err(
-                "Destination already exists: '{}' in '{}'.".format(job, dst_project))
-        else:
-            _print_err("Moved '{}' to '{}'.".format(job, dst_project))
-
-
-def main_clone(args):
-    project = get_project()
-    dst_project = get_project(root=args.project)
-    for job_id in args.job_id:
-        try:
-            job = _open_job_by_id(project, job_id)
-            dst_project.clone(job)
-        except DestinationExistsError:
-            _print_err("Destination already exists: '{}' in '{}'.".format(job, dst_project))
-        else:
-            _print_err("Cloned '{}' to '{}'.".format(job, dst_project))
+@main.command()
+@click.argument('filter', nargs=-1, type=click.STRING)
+@click.option('-d', '--doc-filter', type=click.STRING, cls=MultipleOptionalArgument, nargs='+')
+@click.option('-i', '--index', type=click.STRING)
+@click.option('--show', '-s', is_flag=True)
+@click.option('--sp', cls=MultipleOptionalArgument, type=click.STRING)
+@click.option('--doc', cls=MultipleOptionalArgument, type=click.STRING)
+@click.option('-p', '--pretty', cls=MultipleOptionalArgument, type=click.INT, nargs='?', const=3)
+@click.option('-1', '--one-line', is_flag=True)
+def find(**kwargs):
 
+    for opt in ['filter', 'doc_filter', 'sp', 'doc', 'pretty']:
+        kwargs[opt] = transform_option(kwargs[opt])
 
-def main_index(args):
-    _print_err("Compiling master index for path '{}'...".format(
-        os.path.realpath(args.root)))
-    if args.tags:
-        args.tags = set(args.tags)
-        _print_err("Provided tags: {}".format(', '.join(sorted(args.tags))))
-    for doc in index(root=args.root, tags=args.tags, raise_on_error=args.debug):
-        print(json.dumps(doc))
+    # setting default values
+    kwargs['pretty'] = 3 if kwargs['pretty'] is None else kwargs['pretty']
 
-
-def main_find(args):
     project = get_project()
 
     len_id = max(6, project.min_len_unique_id())
 
     # --show = --sp --doc --pretty 3
     # if --sp or --doc are also specified, those subsets of keys will be used
-    if args.show:
-        if args.sp is None:
-            args.sp = []
-        if args.doc is None:
-            args.doc = []
+
+    if kwargs['show']:
+        kwargs['sp'] = [] if kwargs['sp'] is None else kwargs['sp']
+        kwargs['doc'] = [] if kwargs['doc'] is None else kwargs['doc']
 
     def format_lines(cat, _id, s):
-        if args.one_line:
+        if kwargs['one_line']:
             if isinstance(s, dict):
                 s = json.dumps(s, sort_keys=True)
             return _id[:len_id] + ' ' + cat + '\t' + s
         else:
-            return pformat(s, depth=args.pretty)
+            return pformat(s, depth=kwargs['pretty'])
 
     try:
-        for job_id in find_with_filter(args):
+        for job_id in find_with_filter(**kwargs):
```

Comment:
You might find `types.SimpleNamespace` useful in places that would otherwise require porting between `args` (which is a namespace) and `**kwargs` (which is a dictionary). https://docs.python.org/3/library/types.html#types.SimpleNamespace